src/libostree/ostree-repo-pull.c

```diff
@@ -2018,9 +2083,9 @@ repo_remote_fetch_summary (OstreeRepo    *self,
   glnx_unref_object OstreeFetcher *fetcher = NULL;
   g_autoptr(GMainContext) mainctx = NULL;
   gboolean ret = FALSE;
-  SoupURI *base_uri = NULL;
   gboolean from_cache = FALSE;
   g_autofree char *url_override = NULL;
+  GSList *mirrorlist = NULL;
```

Comment:
Noticing you aren't using autoptr here reminds me...in general, there are really very few *good* use cases for linked lists; you almost always want a vector (`GPtrArray`) instead.  See e.g. [this](http://cglab.ca/~abeinges/blah/too-many-lists/book/#an-obligatory-public-service-announcement).

And with `GPtrArray` we can add a free-element function which makes using `autoptr` possible always.